src/hotspot/share/gc/shenandoah/shenandoahOopClosures.hpp

```diff
@@ -188,41 +188,37 @@ class ShenandoahMarkRefsMetadataDedupClosure : public ShenandoahMarkRefsSuperClo
 };
 
 class ShenandoahUpdateRefsSuperClosure : public BasicOopIterateClosure {
-private:
-  ShenandoahHeap* _heap;
-
 protected:
-  template <class T, UpdateRefsMode UPDATE_MODE>
-  inline void work(T *p);
+  ShenandoahHeap* _heap;
```

Comment:
While moving it around, maybe make it Shenandoah* const too?

Reply:
Right! Did in new changeset.